src/PublicApiGenerator/ApiGenerator.cs

```diff
@@ -675,16 +673,30 @@ static void AddPropertyToTypeDeclaration(CodeTypeDeclaration typeDeclaration, Pr
                 ? new CodeTypeReference(member.PropertyType.Name)
                 : member.PropertyType.CreateCodeTypeReference(member);
 
+            var propertyName = member.Name;
             var property = new CodeMemberProperty
             {
-                Name = member.Name,
+                Name = propertyName,
                 Type = propertyType,
                 Attributes = propertyAttributes,
                 CustomAttributes = CreateCustomAttributes(member, attributeFilter),
                 HasGet = hasGet,
                 HasSet = hasSet
             };
 
+            // DefaultMemberAttribute on type gets propagated to IndexerNameAttribute
+            var defaultMemberAttribute = typeDeclarationInfo.CustomAttributes.SingleOrDefault(x =>
+                x.AttributeType.FullName == "System.Reflection.DefaultMemberAttribute");
+            var defaultMemberAttributeArgument = defaultMemberAttribute?.ConstructorArguments.SingleOrDefault();
+            var value = defaultMemberAttributeArgument?.Value as string;
+            if (!string.IsNullOrEmpty(value) && propertyName != "Item")
```

Comment:
I think this should be `if (value == propertyName)` in case there are two of these things. (VB allows you to declare any number of named indexers, but only one can be the default indexer.)